addons/crm_iap_lead_enrich/models/crm_iap_lead_enrich.py

```diff
@@ -0,0 +1,150 @@
+        if domains:
+            try:
+                response_clearbit = self._make_request(domains)
+                self._enrich_leads_from_response(response_clearbit)
+                self.env['ir.config_parameter'].sudo().set_param('lead_enrich.already_notified', False)
+
+            except InsufficientCreditError:
+                self.notify_no_more_credit('reveal', self._name, 'lead_enrich.already_notified')
+                if cron:
+                    return
+                data = {
+                    'message_title': _("Lead enriched based on email address"),
+                    'url' : self.env['iap.account'].get_credits_url('reveal'),
+                }
+                self.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_no_credit', values=data, subtype_id=self.env.ref('mail.mt_note').id)
+
+    @api.model
+    def enrich_with_cron(self):
+        timeDelta = fields.datetime.now() - datetime.timedelta(hours=1)
+        leads = self.search([('enriched_lead', '=', False),('reveal_id', '=', False),('probability','!=','0'),('probability','!=','100'),('create_date','>',timeDelta)])
+        leads.lead_enrich_mail(True)
+
+    @api.model
+    def _make_request(self, domains):
+        """This method will query the endpoint to get the data for the asked (lead.id, domain) pairs"""
+        reveal_account = self.env['iap.account'].get('reveal')
+        dbuuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = self.env['ir.config_parameter'].sudo().get_param('reveal.endpoint', DEFAULT_ENDPOINT) + '/iap/clearbit/1/lead_enrichment_email'
+        params = {
+            'account_token': reveal_account.account_token,
+            'dbuuid': dbuuid,
+            'domains': domains,
+        }
+        return jsonrpc(endpoint, params=params, timeout=300)
+
+    @api.model
+    def _create_message_data(self, company_data):
+        log_data = {
+            'message_title': _("Lead enriched based on email address"),
+            'twitter': company_data['twitter'],
+            'description': company_data['description'],
+            'logo': company_data['logo'],
+            'name': company_data['name'],
+            'phone_numbers': company_data['phone_numbers'],
+            'facebook': company_data['facebook'],
+            'linkedin': company_data['linkedin'],
+            'crunchbase': company_data['crunchbase'],
+            'tech': [t.replace('_', ' ').title() for t in company_data['tech']],
+        }
+        timezone = company_data['timezone']
+        if timezone:
+            log_data.update({
+                'timezone': timezone.replace('_', ' ').title(),
+                'timezone_url': company_data['timezone_url'],
+            })
+        return log_data
+
+    @api.model
+    def _send_message(self, company_data=False):
+        if company_data:
+            messages_to_post = self._create_message_data(company_data)
```

Comment:
Woudln't it be possible to give the created lead to render in lead_message_template (well in a copy of it, upgraded to work on a record) ? That way you don't have to go through various marshmalling methods that just format data in a way or another. Normally everything you store should be present in the lead and therefore you may give it directly to a qweb template. Do you think this would work ?